tt_metal/hw/inc/debug/sanitize_noc.h

```diff
@@ -30,6 +30,13 @@ enum debug_sanitize_which_riscv {
 
 #if defined(WATCHER_ENABLED)
 
+#if defined(COMPILE_FOR_ERISC)
+// Forward declaration to avoid circular dependency
+namespace internal_ {
+void __attribute__((section("code_l1"))) risc_context_switch();
```

Comment:
isn't this declared elsewhere?  why do we need to re-declare it here?  I see the forward declaration comment, but isn't there another header w/ it?

Reply:
This is to avoid an annoying circular dependency, this file includes `ethernet/dataflow_api.h`, but to use the sanitization functions there `ethernet_dataflow_api.h` needs to include this function. Perhaps going forward we could look to restructure some files to avoid having to do this

Reply:
Filed issue #5452 for this so we don't forget